plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -121,18 +127,22 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
                 override val postconditions = signature.formalArgs.flatMap { it.accessInvariants() } +
                         signature.params.flatMap { it.dynamicInvariants() } +
                         signature.returnVar.invariants() +
+                        signature.returnVar.provenInvariants() +
```

Comment:
This results in these conditions still needing to be proven at the end of the function we're verifying.  I'm not sure how best to avoid this distinction, though...  Perhaps this should be a TODO on when we get method calls done?

(Perhaps a more nuanced encoding of pre-/postconditions, that lets us determine which ones are relevant for what, would be a good idea?  Right now they're all just `Exp`s that we have to treat correctly...)

Reply:
Yes I put them into the post conditions as this actually simplifies things. This is because otherwise you have to inhale these invariants every single time you call a function. Also note that proving them at the end of the function is practically trivial since any function that produces the return value gives you the required post condition automatically.

Thinking about this it might be that field accesses are not handles correctly right now, I'll investigate that.

Maybe there's a nicer way to split up the conditions but I think for now it's totally fine and I would prefer to wait for a more complicated split for when we deal with permission (e.g. for val fields). This might require some extra constraints on the design.

Reply:
Sounds good; I agree that this looks fine for now.